inst/@sym/setxor.m

```diff
@@ -26,39 +26,32 @@
 %% Author: Colin B. Macdonald
 %% Keywords: symbolic
 
-function r = setxor(A, B)
-
-  % FIXME: is it worth splitting out a "private/set_helper"?
-
-  cmd = { 'A, B = _ins'
-          'try:'
-          '    A = iter(A)'
-          'except TypeError:'
-          '    A = set([A])'
-          'else:'
-          '    A = set(A)'
-          'try:'
-          '    B = iter(B)'
-          'except TypeError:'
-          '    B = set([B])'
-          'else:'
-          '    B = set(B)'
-          'C = A.symmetric_difference(B)'
-          'C = sympy.Matrix([list(C)])'
-          'return C,' };
-
-  A = sym(A);
-  B = sym(B);
-  r = python_cmd (cmd, A, B);
-
-  % reshape to column if both inputs are
-  if (iscolumn(A) && iscolumn(B))
-    r = reshape(r, length(r), 1);
-  end
+function r = setxor(a, b)
+
+  %%FIXME: In future version of SymPy, replace (A - B) | (B - A) with A ^ B
+  % Version(spver) >= Version("0.7.7.dev")
+
+  cmd = {
+         'a, b = _ins'
+         'if isinstance(a, sp.Set) or isinstance(b, sp.Set):'
+         '    return (a - b) | (b - a), 1'
+         ''
+         'A = sp.FiniteSet(*(list(a) if isinstance(a, sp.MatrixBase) else [a]))'
+         'B = sp.FiniteSet(*(list(b) if isinstance(b, sp.MatrixBase) else [b]))'
+         'C = (A - B) | (B - A)'
+         'return sp.Matrix([[list(C)]]), 0'
+        };
+
+    [r, out] = python_cmd (cmd, sym(a), sym(b));
+
+    if !out
+      r = horzcat(r{:});
+    end
 
 end
 
 
+
```

Comment:
upstream octave style says exactly two blank spaces after function end and before  tests